src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -517,9 +570,17 @@ class RaftIvfIndexNode : public IndexNode {
                             *res_, search_params, *gpu_index_, raft::make_const_mdspan(data_gpu.view()),
                             ids_gpu_before.view(), dis_gpu_before.view());
 
-                        filter<<<dim3(1, rows), k2, k2 * sizeof(std::int64_t) + k2 * sizeof(float), stream.value()>>>(
-                            k1, k2, rows, bs_gpu.data_handle(), ids_gpu_before.data_handle(),
-                            dis_gpu_before.data_handle(), ids_gpu.data_handle(), dis_gpu.data_handle());
+                        auto ids_block = raft::make_device_vector<int32_t, std::int64_t>(*res_, rows * k2);
+                        filter<<<(rows * k2 + 1023) / 1024, 1024, 0, stream.value()>>>(
+                            bs_gpu.data_handle(), ids_gpu_before.data_handle(), ids_block.data_handle(), rows * k2);
+                        auto pre_ids_block = raft::make_device_vector<int32_t, std::int64_t>(*res_, rows * k2);
+                        RAFT_CUDA_TRY(cudaMemcpyAsync(pre_ids_block.data_handle(), ids_block.data_handle(),
+                                                      rows * k2 * sizeof(int32_t), cudaMemcpyDefault, stream.value()));
+                        prescan<<<dim3(1, rows), k2 / 2, sizeof(int32_t) * k2, stream.value()>>>(
+                            pre_ids_block.data_handle(), ids_block.data_handle(), k2);
+                        write_back<<<dim3(1, rows), k1, sizeof(int32_t) * k2, stream.value()>>>(
+                            pre_ids_block.data_handle(), ids_gpu_before.data_handle(), dis_gpu_before.data_handle(),
```

Comment:
k1 should be limited because threads number could not exceed 1024 for Cuda. Explicitly check this param in config if necessary.